### PR TITLE
Add support for gitlab issue listing

### PR DESCRIPTION
**Pull request by jnduli**

### CODE REVIEW

File: src/Issues.hs [hunk 1]
```diff
@@ -3,65 +3,98 @@
...
-    , title :: String
-    , user :: User
-    , state :: String
-    , created_at :: String
-    , body :: String
-    }
+data Issue
+  = GitlabIssue
+      { title :: String
+      , iid :: Int
```
Comment (BonfaceKilz) [inline src/Issues.hs:39]: Is this a typo? Shouldn't be this `id`?

File: src/Tui.hs [hunk 2]
```diff
@@ -48,7 +48,34 @@ getUrlFromGitRepository = do
...
     True -> return $ AutoUrl $ formGithubUrl url
-    False -> return Fail
+    False ->
+      case "gitlab" `isInfixOf` url of
+        True -> return $ AutoUrl $ formGitlabUrl url
+        False -> return Fail
+
+gitlabIssuesUrlPrefix = "https://gitlab.com/api/v4/projects/"
+
+-- /commento%2Fcommento/issues?state=opened
```
Comment (BonfaceKilz) [inline src/Tui.hs]: NIT: This comment adds no value

Comment (BonfaceKilz) [reply src/Tui.hs]: You should flesh it out to give it more value to the reader IMO

File: src/Tui.hs [hunk 3]
```diff
@@ -48,7 +48,34 @@ getUrlFromGitRepository = do
...
+replaceBackSlash (x:xs) = x : replaceBackSlash xs
+
+-- assumes url is of form 'git@github.com:user/repo.git' or 'git@gitlab.com:user/repo.git`
+giturl :: String -> String
+giturl (':':xs) = removeGitSuffix xs
+giturl (x:xs) = giturl xs
+
+-- assumes url is of form 'https://github.com/user/repo.git
+httpurl ('m':'/':xs) = removeGitSuffix xs
+httpurl (x:xs) = httpurl xs
```
Comment (BonfaceKilz) [inline src/Tui.hs:79]: Since there's an assumption being made, you should use type to keep your programs safe by returning a `Maybe String` instead.

**Review by BonfaceKilz — CHANGES_REQUESTED** [review]
I haven't run this yet, but there's no weirdness that has caught my eye apart from some minor nitpicky changes.

File: src/Issues.hs (hunk 1 above)
Comment (jnduli) [reply src/Issues.hs:39]: gitlab api uses iid for this json field

File: src/Tui.hs (hunk 3 above)
Comment (jnduli) [reply src/Tui.hs:79]: I'll keep this in mind while improving the code. I've added a separate issue for this: https://github.com/jnduli/hidd/issues/4

**Review by BonfaceKilz — APPROVED** [review]
LGTM. I reckon you will address some of the issues I raised in another PR

**Pull request merged**